lib/Locations.php

```diff
@@ -10,23 +10,31 @@
 /**
  * @implements IteratorAggregate<Location>
  */
-class Locations implements IteratorAggregate, Countable
+final class Locations implements IteratorAggregate, Countable
 {
     /**
      * @var Location[]
      */
     private $locations = [];
 
     /**
-     * @param Location[] $locations
+     * @param iterable<Location> $locations
      */
-    public function __construct(array $locations)
+    public function __construct(iterable $locations)
     {
         foreach ($locations as $location) {
             $this->add($location);
         }
     }
 
+    /**
+     * @param iterable<Location> $locations
+     */
+    public static function bySorting(iterable $locations): self
+    {
+        return (new self($locations))->sorted();
+    }
```

Comment:
sorry -  I would remove this as well. It's enough to create a new collection and call `->sorted()`.

"by sorting" doesn't really make sense - `Locations::bySorting`: `by` makes sense for retrieving a set of things (e.g. `usersByFirstName`).

I think `$locations->sorted()` is fine.

Reply:
For me it's obvious with the static call that it is a static method factory, so it means create by sorting.
I could rename it like that if you prefer: `Locations::createBySorting(iterable)`
There is also the option`Locations::sorted(iterable)` but this will require to rename the current `sorted` method back to `sort`.

At the end this is only a helper to avoid having ugly things like: `return (new Locations($locations))->sorted();`

After reading your first review I think you had that in mind from the start so I will remove it right away to speed up things.